flowforge/input/Components.py

```diff
@@ -309,6 +309,148 @@ def factory(indict: Dict) -> Dict[str, Component]:
         return components
 
 
+class CrossSection(abc.ABC):
+    """Abstract base class for cross-sectional areas"""
+
+    @abc.abstractmethod
+    def flow_area(self) -> float:
+        pass
+
+    @abc.abstractmethod
+    def hydraulic_diameter(self) -> float:
+        pass
+
+    @abc.abstractmethod
+    def wetted_perimeter(self) -> float:
+        pass
+
+class CircularCrossSection(CrossSection):
+    """Circular cross-sectional area
+
+    Parameters
+    ---------
+    R: float
+        Radius of the circular pipe
+    """
+
+    def __init__(self, R: float):
+        self._R = R
+
+    @property
+    def R(self) -> float:
+        return self._R
+
+    def flow_area(self) -> float:
+        return np.pi * self._R ** 2
+
+    def hydraulic_diameter(self) -> float:
+        return 2 * self._R
+
+    def wetted_perimeter(self) -> float:
+        return 2 * np.pi * self._R
+
+
+class SquareCrossSection(CrossSection):
+    """Square cross-sectional area
+
+    Parameters
+    ----------
+    W : float
+        Width of the square pipe (i.e. flat-to-flat)
+    """
+
+    def __init__(self, W: float):
+        self._W = W
+
+    @property
+    def W(self) -> float:
+        return self._W
+
+    def flow_area(self) -> float:
+        return self._W**2
+
+    def hydraulic_diameter(self) -> float:
+        return self._W
+
+    def wetted_perimeter(self) -> float:
+        return 4 * self._W
+
+
+class RectangularCrossSection(CrossSection):
+    """Rectangular cross-sectional area
+
+    Parameters
+    ----------
+    W: float
+        Width of the rectangular pipe
+    H: float
+        Height of the rectangular pipe
+    """
+    def __init__(self, W: float, H: float):
+        self._W = W
+        self._H = H
+
+    @property
+    def W(self) -> float:
+        return self._W
+
+    @property
+    def H(self) -> float:
+        return self._H
+
+    def flow_area(self) -> float:
+        return self._W*self._H
+
+    def hydraulic_diameter(self) -> float:
+        return 4 * self.flow_area() / self.wetted_perimeter()
+
+    def wetted_perimeter(self) -> float:
+        return 2 * (self._W + self._H)
+
+
+class StadiumCrossSection(CrossSection):
+    """Stadium cross sectional area
+
+    Parameters
+    ----------
+    A : float
+        Length of the rectangular portion of the stadium channel
+    R : float
+        Radius of the semi cirulcar portion of the stadium channel
+    """
+    def __init__(self, A: float, R: float):
+        self._A = A
+        self._R = R
+
+    @property
+    def A(self) -> float:
+        return self._A
+
+    @property
+    def R(self) -> float:
+        return self._R
+
+    def flow_area(self) -> float:
+        return np.pi * (self._R)**2 + 2 * (self._R * self._A)
```

Comment:
ok, this is  nitpicky, but the area of the rectangular region should really be thought of a `(2 * self._R) * self._A`.  It's based times height where the based is `self._A` and height it `(2 * self._R)`.  In all honesty, I would just write it as `2 * self._R * self._A`